tests/garage/tf/optimizers/test_conjugate_gradient_optimizer.py

```diff
@@ -0,0 +1,87 @@
+"""Tests for garage.tf.optimizers.conjugateGradientOptimizer"""
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.optimizers.conjugate_gradient_optimizer import (
+    cg, FiniteDifferenceHvp, PerlmutterHvp)
+from garage.tf.policies import Policy
+
+
+class OneParamPolicy(Policy):
+    """Helper class for testing hvp classes"""
+
+    def __init__(self, name='OneParamPolicy'):
+        super().__init__(name, None)
+        with tf.compat.v1.variable_scope(self.name) as vs:
+            self._variable_scope = vs
+            _ = tf.Variable([0.])
+
+    def get_action(self, observation):
+        pass
+
+    def get_actions(self, observations):
+        pass
+
+
+class TestConjugateGradientOptimizer:
+    """Test class for ConjugateGradientOptimizer and HVP classes"""
+
+    def test_cg(self):
+        """Solve Ax = b using Conjugate gradient method."""
+        a = np.random.randn(5, 5)
+        a = a.T.dot(a)  # make sure a is positive semi-definite
+        b = np.random.randn(5)
```

Comment:
please don't use random data in tests. you can instead use well-known sequences like pi, e, fibonacci, etc.

Reply:
The cg function should solve x for any equation `Ax = b` as long as A is a positive semi definite matrix. So, it should pass regardless of the values of `A` and `b`.

Reply:
Sorry, I didn't meant to imply the test is wrong. I'm sure it works.

We don't use RNGs in the testing suite ever, because makes tests non-deterministic.

Reply:
I see, that makes sense.